tree_test.go

```diff
@@ -0,0 +1,210 @@
+package rsmt2d
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+// TestRegisterTree tests the RegisterTree function for adding
+// a tree constructor function for a given tree name into treeFns
+// global map, consists of 2 test cases:
+// - The tree has not been registered yet in the treeFns global map
+// in which result in the constructor fn for the new tree type being
+// added to the global map.
+// - The tree has already been registered in the treeFns global map
+// in which result in an error returned.
+func TestRegisterTree(t *testing.T) {
+	treeName := "testing_register_tree"
+	treeConstructorFn := sudoConstructorFn
+
+	tests := []struct {
+		name      string
+		expectErr error
+	}{
+		{"register successfully", nil},
+		{"register unsuccessfully", fmt.Errorf("%s already registered", treeName)},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			// By registering the function on the successful testcase first
+			// the tree name will be registered already so we can check
+			// the unsuccessful testcase.
```

Comment:
This comment applies to a line that is invoked for both test cases so it seems misleading.

```suggestion
```